ghedesigner/ghe/gfunction.py

```diff
@@ -52,45 +46,48 @@ def calculate_g_function(
         case _:
             raise ValueError(f"bhe_type {bhe_type} is not supported")
 
-    pyg_ghe = PyGHE(
-        borehole.H,
-        borehole.D,
-        borehole.r_b,
-        [x for x, _ in coordinates],
-        [y for _, y in coordinates],
-        PyPipeType[pipe_map[bhe_type.name]],
-        pipe.pos,
-        r_inner,
-        r_outer,
-        soil.k,
-        grout.k,
-        pipe.k,
-        fluid.name,
-        fluid.concentration_percent,
-        m_flow_borehole * (len(coordinates)),
-        pipe.roughness,
-    )
-
     # setup options
-    segments = segments.lower()
-    if segments == "equal":
-        options = {"nSegments": n_segments, "disp": disp}
-    elif segments == "unequal":
-        if segment_ratios is None:
-            segment_ratios = gt.utilities.segment_ratios(n_segments, end_length_ratio=end_length_ratio)
-        options = {
-            "nSegments": n_segments,
-            "segment_ratios": segment_ratios,
-            "disp": disp,
-        }
-    else:
-        raise ValueError(f"Unknown segments: {segments}")
-
-    gfunc = pyg_ghe.evaluate_g_function(
-        soil.alpha, time_values, method=solver, boundary_condition=boundary, options=options
+    # none of these were ever used or even exposed for users to access them. hardcoding them here until needed.
+    solver = "equivalent"
+    disp = False
+    n_segments = 8
+    end_length_ratio = 0.02
+
+    options = {
+        "nSegments": n_segments,
+        "segment_ratios": gt.utilities.segment_ratios(nSegments=n_segments, end_length_ratio=end_length_ratio),
+        "disp": disp,
+    }
+
+    nbh = len(coordinates)
+    m_flow_network = nbh * m_flow_borehole
+
+    g_func = gFunction.from_static_params(
```

Comment:
Except from passing `Borehole` instances all over the place, this should now be one of our only touch points with pygfunction.